lib/fastlane/actions/s3.rb

```diff
@@ -261,7 +261,9 @@ def self.get_ipa_info(ipa_file)
         bundle_id, bundle_version, title, full_version = nil
         Dir.mktmpdir do |dir|
           system "unzip -q #{ipa_file} -d #{dir} 2> /dev/null"
-          plist = Dir["#{dir}/**/*.app/Info.plist"].last
+          plist = Dir["#{dir}/**/*.app/Info.plist"].detect do|f|
+            !f.match %r{Payload/(?:.+)\.app/.+\.app/Info.plist$}
```

Comment:
Would you mind moving this into `fastlane_core`? We already have some code here to retrieve this kind of data, and your solution looks pretty awesome https://github.com/fastlane/fastlane_core/blob/master/lib/fastlane_core/ipa_file_analyser.rb#L22